### PR TITLE
Update typescript grammar and queries

### PR DESCRIPTION
**Pull request by gabydd**
fixes: #4702
updates grammars for typescript, and tsx
as well as updates queries for template literal types

### CODE REVIEW

File: runtime/queries/typescript/highlights.scm [hunk 1]
```diff
@@ -62,3 +62,15 @@
 
 ((identifier) @type
  (#match? @type "^[A-Z]"))
+
```
Comment (archseer) [inline runtime/queries/typescript/highlights.scm:65]: `"override"` was also added to the list of keywords https://github.com/tree-sitter/tree-sitter-typescript/compare/3e897ea5925f037cfae2e551f8e6b12eec2a201a..6aac031ad88dd6317f02ac0bb27d099a553a7d8c#diff-d485a982e458ef8da2cc203585065b7542665cb80b78d230b1e8f77ea25825d4R33

Comment (gabydd) [reply runtime/queries/typescript/highlights.scm:66]: just pushed a fix, thanks Blaz!

File: runtime/queries/typescript/highlights.scm [hunk 2]
```diff
@@ -34,6 +34,7 @@
   "implements"
   "keyof"
   "namespace"
+  "abstract"
```
Comment (archseer) [inline runtime/queries/typescript/highlights.scm:37]:
```suggestion
  "override"
```

Comment (gabydd) [reply runtime/queries/typescript/highlights.scm]: Lol sorry about that

**Pull request merged**